sites/topomath.asu.edu/modules/system_descriptions/static/sys_desc.js

```diff
@@ -0,0 +1,261 @@
+		}
+		$.post(sd_update_handler, input)
+			.success(function (datum) {
+				console.log("received",datum);
+				var sys_json = JSON.parse(datum);
+				for(var sdName in sys_json){
+					console.log(sdName, sys_json[sdName]["path"], sys_json[sdName]["type"]);
+					var descName = sdName.trim();
+					console.log("name", descName);
+					if(descName!= ""){
+						var descpath = sys_json[sdName]["path"].trim();
+						var descLink = uploaddir+descpath;
+						var descType = sys_json[sdName]["type"].trim();
+						console.log(" descType is",descType);
+						if(descType == "cketext")
+							$('#list-sys-descriptions').append("<div><a href = '"+descLink+"' style='cursor: pointer' target='_blank'>"+descName+"</a><span class='upload_glyph glyphicon glyphicon-edit edit-sys-desc-inline' data-syspath='"+descpath+"'></span><span class='upload_glyph glyphicon glyphicon-remove delete-sys-desc-inline' data-syspath='"+descpath+"'></span></div><br/>");
+						else if(descType == "upload")
+							$('#list-sys-descriptions').append("<div><a href = '"+descLink+"' style='cursor: pointer' target='_blank'>"+descName+"</a><span class='upload_glyph glyphicon glyphicon-remove delete-sys-desc-inline' data-syspath='"+descpath+"'></span></div><br/>");
+					}
+				}
+		});
+
+	};
+
+	/* add new description
+	
+	*/
+
+	var addNewDescription = function(fileName,pathName){
+		//step 1: collect data relevant to the operations of this module
+		var form = document.forms['dragoon_problem_form'];
+		var user = form["u"].value;
+		var folder = form["f"].value;
+		var problem = form["p"].value;
+		var section = form["s"].value;
+		var name = fileName;
+		var path = pathName;
+		var input = {
+			u: user,
+			f: folder,
+			p: problem,
+			s: section,
+			n: name,
+			path: path,
+			req_type: "newDesc",
+			desc_type: "upload"
+		}
+		$.post(sd_update_handler, input)
+			.success(function (datum) {
+				console.log("received",datum);
+				//location.reload();
+		});		
+	};
+
+	/*
+		File name format, name duplication and file size errors
+	*/
+	var basicFilechecks = function(file){
+		$('#sysDesc-error-box').html('');
+
+		//special characters case
+        var pattern = new RegExp(/[~`!#$%\^&*+=\-\[\]\\';,/{}|\\":<>\?]/);
+        if (pattern.test(file.name)) {
+        	console.error("special characters not allowed in file name");
+        	$('#sysDesc-error-box').html('File name cannot contain special characters');
+        	$('.progress-bar').hide();
+        	return false;
+        }
+		// file size error
+		if( (file.size/(1024*1024)) > 25 ){
+			console.error("please limit file upload size");
+			$('#sysDesc-error-box').html('File upload size cannot exceed 25 mb');
+			$('.progress-bar').hide();
+			return false;
+		}
+		if(file.type !== "application/pdf"){
+			console.error("file format not supported");
+			$('#sysDesc-error-box').html('Only pdf files are allowed as uploads');
+			$('.progress-bar').hide();
+			return false;
+		}
+		return true;
```

Comment:
Yeah this string comes from checkDuplication function which prints the string. (I think you got confused because I used same variable name which I will change)